File: src/main/java/instance/common/DiskChannel.java
package instance.common;

import instance.disk.StartDiskUnit;
import instance.os.RestartSignal;
import se.sics.kompics.PortType;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-18
 *
 */

public class DiskChannel extends PortType {
	{
		positive(Ready.class);
		positive(BlockResponse.class);
		negative(LoadBlock.class);
		negative(ReadBlock.class);
		negative(RestartSignal.class);
		negative(StartDiskUnit.class);
	}
}

File: src/main/java/instance/common/DiskChannel.scala
package instance.common

import se.sics.kompics.PortType
import instance.os.RestartSignal
import instance.disk.StartDiskUnit

/**
 * @author Amir Moulavi
 *
 */
class DiskChannel extends PortType {

  positive(classOf[Ready])
  positive(classOf[BlockResponse])
  negative(classOf[LoadBlock])
  negative(classOf[ReadBlock])
  negative(classOf[RestartSignal])
  negative(classOf[StartDiskUnit])

}